### DISCUSSION TITLE
SVG Rendering

### DISCUSSION THREAD

**Discussion by ryanbateman** [openrndr/openrndr#390] [General]
Hello.

I'm interested in improving OpenRNDR's SVG support, with a personal end-goal being to be able to render SVG's exported from OpenStreetMap layers. As far as I can see, the current `SVGLoader` class supports only the 'tiny' base profile and so far not fully. How can I best help in increasing SVG support in line with any current goals or work on this? Failing any specific requirements, I'd like start improving just 'tiny' profile support.

Thanks!

**Comment by hamoid** [comment]
Maybe a good start could be to add links and information here as comments? For example links showing what is currently implemented, what is missing from the 'tiny' profile, how large is the 'full' profile... I believe one thing currently missing is text rendering. I think so far you @ryanbateman, @ricardomatias, @Yvee1 and I @hamoid have made small contributions to the SVG area.

Are these the right specs of SVG Tiny? 
https://www.w3.org/TR/SVGTiny12/single-page.html

**Comment by Yvee1** [comment]
I've worked on parsing a list of transforms in #178, and I think that that works now and is complete (at least the 2D transformations, not the 3D ones, I used [this mozilla page](https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/transform) as reference), but I haven't tested it much.

Some missing things I've encountered with the svg reader so far are:
1. Parsing styles in group nodes. E.g. the svg of the greek delta that I put in the tests ([link](https://github.com/openrndr/openrndr/blob/master/openrndr-svg/src/test/resources/svg/delta.svg)) has a white rectangle as background, but this is parsed/drawn by openrndr as a black rectangle currently. So I think `style="fill:white; stroke:white"` is not parsed or used.
2. I also think `documentBounds` is never assigned anything other than the default value. But I discovered there's also `root.bounds` which is probably what you want in most cases

**Comment by hamoid** [comment]
Related issue: https://github.com/openrndr/openrndr/issues/131 (Add support to viewBox attribute)

**Comment by hamoid** [comment]
See https://github.com/openrndr/openrndr/pull/227 (Greatly improve SVG support)
